CMakeLists.txt

```diff
@@ -218,11 +218,16 @@ find_package(Boost REQUIRED COMPONENTS
 )
 message(STATUS "Linking Boost libraries: ${Boost_LIBRARIES}")
 
-### LLVM and Clang - Preferring Clang 15
-find_package(LLVM 15 REQUIRED CONFIG)
+### LLVM and Clang
+find_package(LLVM ${LLVM_REQUESTED_VERSION} REQUIRED CONFIG)
```

Comment:
Here's a nice little feature you could copy from bpftrace to make it easier for new users to set up. Make `LLVM_REQUESTED_VERSION` optional so they won't have to pass `-LLVM_REQUESTED_VERSION` if they only have one version installed:
```
if(${LLVM_REQUESTED_VERSION})
  find_package(LLVM ${LLVM_REQUESTED_VERSION} REQUIRED CONFIG)
else()
  find_package(LLVM REQUIRED CONFIG)
endif()
```

Reply:
Oh, cool!